src/h_vialib/client.py

```diff
@@ -56,7 +57,10 @@ def __init__(self, secret, service_url=None, html_service_url=None):
             "via.external_link_mode": "new-tab",
         }
 
-    def url_for(self, url, content_type=None, options=None, blocked_for=None):
+    # pylint:disable=too-many-arguments
+    def url_for(
+        self, url, content_type=None, options=None, blocked_for=None, headers=None
+    ):
```

Comment:
`headers` are only supported by certain Via endpoints, right? The ones that proxy in Python, not the ones that proxy in NGINX. Are there some `url`'s where `url_for(url, headers=my_headers)` will return a URL containing headers that Via will actually ignore because it's going to proxy that URL through NGINX?

Reply:
> will return a URL containing headers that Via will actually ignore because it's going to proxy that URL through NGINX?

Yes, all HTML ones and pdfs not proxied by python (so all of them except one drive and d2l ones).

Applying the headers to everything will be a big tasks (not sure even if possible without writing nginx lua code).

Maybe we could just explain the limitation on the method's docstring?

Reply:
Added a comment about this. Not ideal but naming this `headers_for_python_proxing` is not better.

Reply:
Maybe we could make Via throw some sort of "not implemented" error if any `headers` are passed to it and it would have proxied the request through NGINX rather than through Python?